apps/server/src/modules/lesson/lesson.module.ts

```diff
@@ -1,16 +1,26 @@
 import { Module } from '@nestjs/common';
 import { FeathersServiceProvider } from '@shared/infra/feathers';
-import { LessonRepo } from '@shared/repo';
+import { CourseRepo, LessonRepo } from '@shared/repo';
 import { LoggerModule } from '@src/core/logger';
 import { CopyHelperModule } from '@src/modules/copy-helper';
 import { FilesStorageClientModule } from '@src/modules/files-storage-client';
 import { TaskModule } from '@src/modules/task';
+import { CourseService } from '../learnroom/service/course.service';
 import { EtherpadService, LessonCopyService, LessonService, NexboardService } from './service';
 
 @Module({
 	imports: [FilesStorageClientModule, LoggerModule, CopyHelperModule, TaskModule],
 	controllers: [],
-	providers: [LessonRepo, LessonService, EtherpadService, NexboardService, LessonCopyService, FeathersServiceProvider],
+	providers: [
+		LessonRepo,
+		LessonService,
+		EtherpadService,
+		NexboardService,
+		LessonCopyService,
+		FeathersServiceProvider,
+		CourseService,
+		CourseRepo,
```

Comment:
we have a problem here...
now we have the courseservice and repo in the lesson module, which breaks the rule that they should only be provided in their own module (learnroom). And the reason is, as I saw, that you need to pass a course into the lesson constructor (and importing the learnroom module here leads to a dependency cycle).

the long term solution is the domain object rework. until then, we have to find another solution.
the second solution would be to make the constructor of the lesson independent from a course entity, which it should be (based on the id). I remember there was a way to do that, but that requires digging in mikroorm internals and changing the lesson entity constructor, and thats a bit heavy for your first nestjs feature. So I wont ask you to do that.
My final solution idea would be, (and im very sorry for the back and forth), go back to passing the course (but not the lesson) as entity instead of id. Im sorry for the back and forth, but im gonna ask you to try that one.

That means the lesson copy uc and the sharetokenuc both need to use the CourseService to fetch the course, and pass it to the LessonCopyService. as far as I can see, both should have access to the CourseService already.

Reply:
Lesson.module.ts now looks like it did before my commits